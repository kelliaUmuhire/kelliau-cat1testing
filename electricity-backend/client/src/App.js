import logo from "./logo.svg";
// import "./App.css";
import GetToken from "./components/GetToken/GetToken";
import { useState } from "react";
import CheckToken from "./components/CheckToken/CheckToken";

function App() {
  const [pageI, setPageI] = useState(0);
  return (
    <div className="App">
      <h1>Welcome</h1>
      {/* <h1 className="text-red-500">Hello</h1> */}

      {pageI === 1 ? (
        <GetToken />
      ) : pageI === 2 ? (
        <CheckToken />
      ) : (
        <div>
          <button onClick={() => setPageI(1)}>Get Token</button>
          <button onClick={() => setPageI(2)}>Check Token</button>
        </div>
      )}
    </div>
  );
}

export default App;
