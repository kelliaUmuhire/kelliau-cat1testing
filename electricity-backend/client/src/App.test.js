import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

test("renders buttons", () => {
  render(<App />);
  const btn1 = screen.getByText("Get Token");
  const btn2 = screen.getByText("Check Token");
  expect(btn1).toBeInTheDocument();
  expect(btn2).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// import React from "react";
// import { mount } from "enzyme";
// import App from "./App";

// it("renders correctly", () => {
//   const wrapper = mount(<App />);
//   expect(wrapper.state("error")).toEqual(null);
// });
