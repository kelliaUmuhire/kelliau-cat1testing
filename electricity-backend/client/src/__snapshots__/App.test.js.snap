// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Welcome
  </h1>
  <div>
    <button
      onClick={[Function]}
    >
      Get Token
    </button>
    <button
      onClick={[Function]}
    >
      Check Token
    </button>
  </div>
</div>
`;
